smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/SwiftWriter.kt

```diff
@@ -154,7 +154,13 @@ class SwiftWriter(
         // Package.swift requires a special comment at the top to specify Swift tools version,
         // and the package manifest generator manually writes its own dependency imports
         // (it only imports the PackageDescription module.)
-        return contents.takeIf { fullPackageName == "Package" } ?: (copyrightNotice + imports + contents)
+        //
+        // Also leave out the headers when JSON or the version file is being written,
+        // as indicated by the file extension.
+        val isPackageManifest = listOf(PACKAGE_MANIFEST_NAME, (PACKAGE_MANIFEST_NAME + ".swift")).contains(fullPackageName)
```

Comment:
Q: Why is `fullPackageName` (which is `"Package"` for a Swift Package.swift file) compared against `"Package.swift.txt"` and `"Package.swift.txt.swift"`? Can't we just delete this `val` declaration and do the following?
```
return contents.takeIf { fullPackageName == "Package" && isNonSwiftSourceFile } ?: (copyrightNotice + imports + contents)
```

Reply:
If you look up above, the suffix `.swift` is stripped off of `fullPackageName` (which itself isn't a very good name).  I don't know the precise rationale for this, but for now I don't want to alter that behavior.

In the interim, until we go all-in on individual service client packages, I want to generate the package manifests but not have Xcode treat the services as their own packages, so I changed `PACKAGE_MANIFEST_NAME` to `Package.swift.txt` so the file is still there but it's not recognized as a package manifest.

But since I want `Package.swift.txt` to omit the headers, same as `Package.swift`, I had to include two alternate strings on Line 160 so I don't have to change `PACKAGE_MANIFEST_NAME` in more than one place to make the switch correctly.

This logic will all go away once service client packages are permanent.